### ISSUE TITLE
Non-stable training

### ISSUE THREAD

**Issue by leobol96** [Unity-Technologies/ml-agents#2093]
Hi, there, 

These are the specifications of my Environment. 

As you can see in the attached image is composed of 8 cubes.
The agent (the sphere) must take one of the cubes below and bring it near the cube of the same color among those present in the above floor.

![env](https://user-images.githubusercontent.com/36326039/58694420-3ee2e880-8393-11e9-97c0-1a6687aedcc4.PNG)

The agent is rewarded with :
0.125 for each cube taken from those below
0.250 for each cube brought close to the corresponding 

No more cubes can be taken at a time.
The cubes in the upper floor cannot be taken.
The cubes in the floor below are created randomly.
The agent cannot go out beyond a certain zone because there are masks.
The maximum reward is 1.50, and the maximum steps are 4096.

![train1 (2)](https://user-images.githubusercontent.com/36326039/58694909-68504400-8394-11e9-95f1-2f2c1645b980.PNG)

![train1_e](https://user-images.githubusercontent.com/36326039/58698023-ba946380-839a-11e9-9352-33fd21dfa51d.PNG)

![train2 (2)](https://user-images.githubusercontent.com/36326039/58694912-6ab29e00-8394-11e9-9d0f-487c969d606b.PNG)

![train2_e](https://user-images.githubusercontent.com/36326039/58698041-c5e78f00-839a-11e9-8cf3-46cc10c639b3.PNG)

![train3 (2)](https://user-images.githubusercontent.com/36326039/58694915-6dad8e80-8394-11e9-836a-12f19c4a658f.PNG)

![train3_e](https://user-images.githubusercontent.com/36326039/58698069-d861c880-839a-11e9-95a1-263f30fc66a1.PNG)

As you can see from the graphs I have problems with my training, when the agent seems to have understood how to get the maximum reward, there is an incredible collapse of the reward and an increase in entropy.
This happened in all the training I did, even though I changed the parameters of the config and entered a negative reward for each step.

This is the config file for the blue and the orange trainining:
 
    trainer: ppo
    batch_size: 2048
    beta: 1.0e-4
    buffer_size: 131072
    epsilon: 0.2
    gamma: 0.99
    hidden_units: 512
    lambd: 0.95
    learning_rate: 5.0e-4
    max_steps: 8.0e6
    memory_size: 256
    normalize: false
    num_epoch: 5
    num_layers: 3
    time_horizon: 1024
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    use_curiosity: false
    curiosity_strength: 0.01
    curiosity_enc_size: 128

And I used : 
--num-envs=16

I read that normalizing the observations could help, is that it? 
Could there be a flaw in the project?

**Comment by DroneMesh** [comment]
Make more platforms and do not use --num-envs it is unreliable I have the same issues when I run num-envs..

I have confirmed this today through testing.
I was searching to see if anyone also had issues with that parameter. 

Keep me updated if that helps

**Comment by leobol96** [comment]
@DroneMesh @unityjeffrey 

Hi , I have already finished some tests on my environment. 
These are my results :

- Same config file 
- 8 milion steps per train
- Grey = num-envs = 1 
- Green = num-envs = 16 

![issue](https://user-images.githubusercontent.com/36326039/58828356-8b554f00-8645-11e9-868c-2f8636d07595.PNG)
![issue_en](https://user-images.githubusercontent.com/36326039/58828364-901a0300-8645-11e9-86b6-dc9cc4bb2952.PNG)

I think that there is a bug with the latest version of mlagents when num-envs is greater than 1.

Keep me updated on the situation

**Comment by roboserg** [comment]
I have something similar - [https://puu.sh/DNO6q/bfd64dea7f.png](picture)

**Comment by leobol96** [comment]
@roboserg I think they fixed the bug with the new realease. Did you update ml-agent ?

**Comment by ervteng** [comment]
Thanks for the discussion - closing this issue due to inactivity. Feel free to open a new issue if you're still running into problems.
